.github/workflows/docker-build.yaml

```diff
@@ -75,7 +75,7 @@ jobs:
         password: ${{ env.REGISTRY_PASSWORD }}
 
     - name: Build and push ${{ inputs.name }} image
-      uses: docker/build-push-action@v3
+      uses: docker/build-push-action@v4
```

Comment:
I think since you added the `provenance` option it shouldn't break our workflows but I rather double check with you @Emylou-s to know what you think.

Reply:
Given the changelog, it's a duplication of v3.3.0, since indeed we swiftly added the provenance to false, V4 won't be a pb.

Funnily though, v3.3.1, they disabled it, by default, but we were faster in solving our own problems ^^
I guess they realized this is the kind of breaking change that needs to go onto a major version :) 

Approved for me

@tcarmet can you teach (or point) me how to add this kind of dependabot for our others repos ? otherwise we will end up in being late to the party and only bump if we get deprecation messages in our PRs.

Reply:
Yeah I saw that revert release they did, very funny.

Sounds good to me let's merge.

I have a quick example on [devdocs](https://devdocs.scality.net/ci-engines/github-actions/best-practices/) you should find better ones or more info on the doc of [dependabot](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file)

Or have a look at the PR #12 which is the one that does the trick.

Reply:
Worth noting that there may be extra work to ensure that the bot will have the right permissions to work, as by default workflows triggered by the bot have read only access.

Per example I had to fix one workflow here #18

Reply:
You reminded me that it would be good to document that. [RELENG-7190](https://scality.atlassian.net/browse/RELENG-7190)